scipy/cluster/hierarchy.py

```diff
@@ -127,13 +134,14 @@
 import numpy as np
 from . import _hierarchy, _optimal_leaf_ordering
 import scipy.spatial.distance as distance
+from scipy._lib._disjoint_set import DisjointSet
 
 
 _LINKAGE_METHODS = {'single': 0, 'complete': 1, 'average': 2, 'centroid': 3,
                     'median': 4, 'ward': 5, 'weighted': 6}
 _EUCLIDEAN_METHODS = ('centroid', 'median', 'ward')
 
-__all__ = ['ClusterNode', 'average', 'centroid', 'complete', 'cophenet',
+__all__ = ['ClusterNode', 'DisjointSet', 'average', 'centroid', 'complete', 'cophenet',
```

Comment:
Just to save a bit of time while waiting for the next batch of CI results: [Linter thinks](https://travis-ci.org/github/scipy/scipy/jobs/724697851) this line is too long.

Reply:
Fixed.